src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -313,39 +339,31 @@ public override ContextMenuStrip ContextMenuStrip
         public Func<ObjectId?, string>? DescribeRevision { get; set; }
 
         public bool FilterFocused => FilterComboBox.Focused;
-        public bool SearchFocused => SearchComboBox.Focused;
+        public bool SearchFocused => SearchCommitGitGrepComboBox.Focused;
 
         /// <summary>
-        /// Enable Search Commit combobox for the file list.
-        /// Refresh of list must be called explicitly.
+        ///  Indicates whether the git-grep search functionality is enabled for this control.
         /// </summary>
-        public bool SearchEnabledForList
-        {
-            private get => _searchEnabledForList;
-            set
-            {
-                _searchEnabledForList = value;
-                EnableSearchForList(value && AppSettings.ShowSearchCommit.Value);
```

Comment:
Removing this works due to CanUseSearchCommitGitGrep is called after FileStatusList is created and init (without the box), then UICommands_PostSettings happens to be called the first time FileStatusDiffCalculator.GetModule() is called.
Is this reliable?

Reply:
Good point. Added an explicit call to `DiffFiles.SetSearchCommitGitGrepVisibility` in `RevisionDiffControl..ctor`.